app/Domains/Vault/ManageReports/Web/ViewHelpers/ReportAddressIndexViewHelper.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace App\Domains\Vault\ManageReports\Web\ViewHelpers;
+
+use App\Models\Vault;
+use Illuminate\Support\Str;
+
+class ReportAddressIndexViewHelper
+{
+    public static function data(Vault $vault): array
+    {
+        // all the cities in the vault
+        // the distinct method does not do a case-insensitive search, so we need
+        // to do it manually. there would be a way to do it with raw SQL but
+        // it wouldn't work on sqlite
+        $cities = $vault->addresses()
+            ->select('id', 'city')
+            ->whereNotNull('city')
+            ->withCount('contacts')
+            ->distinct('city')
+            ->get()
+            ->map(fn ($address) => [
+                'id' => $address->id,
+                'name' => Str::ucfirst($address->city),
+                'contacts' => $address->contacts_count,
+                'url' => [
+                    'index' => route('vault.reports.addresses.cities.show', [
+                        'vault' => $vault->id,
+                        'city' => urlencode(utf8_encode($address->city)),
```

Comment:
I don't think `urlencode` is necessary with `route` method.

Reply:
It is unfortunately.